docker/ci/build

```diff
@@ -4,6 +4,29 @@ set -eux
 
 BUNDLER_PATH=.bundle/gems
 
+function build_and_test_image() {
+  container_name=$1
+
+  : ===
+  : === run linter on the docker files
+  : ===
+   bundle exec puppet-docker local-lint $container_name
+
+  : ===
+  : === build and test $container_name
+  : ===
+  bundle exec puppet-docker build $container_name --repository pcr-internal.puppet.net/release-engineering
```

Comment:
Should we concerned about community members being able to build?

Reply:
At least in it's current state, this script is intended entirely for running on CI. It may be valuable at some point to have this all environment variable'd out, but I think for right now I think community members running the `puppet-docker` commands individually is reasonable